packages/@aws-cdk/assert/lib/inspector.ts

```diff
@@ -63,7 +63,7 @@ export class StackPathInspector extends Inspector {
     // then try with the stack name preprended for backwards compat with most tests that happen to give
     // their stack an ID that's the same as the stack name.
     const metadata = this.stack.manifest.metadata || {};
-    const md = metadata[this.path] || metadata[`/${this.stack.name}${this.path}`];
+    const md = metadata[this.path] || metadata[`/${this.stack.id}${this.path}`];
```

Comment:
Not 100% sure about this one. In most cases `name` and `id` are the same unless `stackName` is specified.

Reply:
Also not if the stack is a child of another construct, right? Or is id here not nodeid but uniqueid?

Reply:
Oh you are right. I think it should be `artifactId` actually.

Reply:
that's artifact ID

Reply:
It's not `core.Stack` it's `CloudFormationStackArtifact`

Reply:
Oh it's for tests too.

Our metadata paths are a little underspecified, it was pretty annoying when I was refactoring tests :(